src/exchange.rs

```diff
@@ -2,11 +2,46 @@ use crate::currency::FormattableCurrency;
 use crate::{Money, MoneyError};
 use rust_decimal::Decimal;
 use std::collections::HashMap;
+use std::hash::{Hash, Hasher};
+use std::fmt;
+use rust_decimal_macros::*;
+
+struct ExchangeRateQuery<'a, T: FormattableCurrency>{
```

Comment:
What do you feel is the primary benefit of using a struct vs just passing in the currencies and letting Exchange handle the complexity?

Reply:
Admittedly, I mostly did this is to play around with implementing keys for HashMap since I'm still new to Rust; however, I do think using a dedicated type is more idiomatic and feels more ergonomic. That being said, if you want to switch over to using string keys that's fine by me.

Reply:
If we can stick with using the strings and the same interface and avoiding a new type, we can merge the change safely into the 0.4.2 release without worrying about backward compat fixes. 

i'm rethinking the exchange design for the next big release (0.5), so perhaps we can revisit this idea then.